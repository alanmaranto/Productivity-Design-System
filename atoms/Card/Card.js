import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import { options } from "./constants";
import styles from "./Card.module.css";

const Card = ({ color, size, onClick, children }) => {
  return (
    <div
      onClick={onClick}
      className={classNames(styles.card, {
        [styles[`color-${color}`]]: color,
        [styles[`size-${size}`]]: size,
        [styles["is-clickable"]]: !!onClick,
      })}
    >
      {children}
    </div>
  );
};

Card.defaultProps = {
  color: "base",
  size: "sm",
};

Card.propTypes = {
  color: PropTypes.oneOf(options.colors),
  size: PropTypes.oneOf(options.sizes),
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
};

export default Card;
