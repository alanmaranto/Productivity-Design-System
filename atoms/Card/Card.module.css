.card {
  width: 100%;
  /* max-width: 320px; */
  border-radius: var(--border-radius-sm);
  box-shadow: var(--box-shadow-sm);
  padding: 12px 10px;
  color: var(--color-font-base);
  display: flex;
  align-items: center;
  border: var(--border-width-thin) solid var(--color-base-transparent);
}

.size-sm {
  height: 45px;
}

.size-md {
  height: 80px;
}

.size-lg {
  height: 115px;
}

.color-primary {
  background: var(--color-primary);
  color: var(--color-primary-inverted);
}
.color-secondary {
  background: var(--color-secondary);
  color: var(--color-secondary-inverted);
}
.color-base {
  border: var(--border-width-thin) solid var(--color-gray-300);
  background: var(--color-primary-inverted);
}
