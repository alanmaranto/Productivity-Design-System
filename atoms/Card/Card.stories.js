import Card, { options } from ".";
import {
  getOptionsArgTypes,
  getTemplate,
  getListTemplate,
} from "../../helpers/storybook";

export default {
  title: "Atoms/Card",
  component: Card,
  args: {
    children: "Laceratons coaster sort comings windlance happily Elf-witch",
  },
  argTypes: {
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.sizes),
  },
};

const Template = getTemplate(Card);
const ListTemplate = getListTemplate(Card);

export const Default = Template.bind({});

export const Colors = ListTemplate.bind({});
Colors.args = {
  items: options.colors.map((color) => ({ color })),
};

export const Sizes = ListTemplate.bind({});
Sizes.args = {
  items: options.sizes.map((size) => ({ size })),
};
