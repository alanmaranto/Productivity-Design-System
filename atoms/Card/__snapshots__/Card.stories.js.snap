// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Card Colors 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="card color-base"
  >
    Laceratons coaster sort comings windlance happily Elf-witch
  </div>
  <div
    className="card color-primary"
  >
    Laceratons coaster sort comings windlance happily Elf-witch
  </div>
  <div
    className="card color-secondary"
  >
    Laceratons coaster sort comings windlance happily Elf-witch
  </div>
</div>
`;

exports[`Storyshots Atoms/Card Default 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="card"
  >
    Laceratons coaster sort comings windlance happily Elf-witch
  </div>
</div>
`;

exports[`Storyshots Atoms/Card Sizes 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="card size-sm"
  >
    Laceratons coaster sort comings windlance happily Elf-witch
  </div>
  <div
    className="card size-md"
  >
    Laceratons coaster sort comings windlance happily Elf-witch
  </div>
  <div
    className="card size-lg"
  >
    Laceratons coaster sort comings windlance happily Elf-witch
  </div>
</div>
`;
